### ISSUE TITLE
No versions of 'googleappmeasurement' match the requirement 10.23.0 and 'firebase-ios-sdk' depends on 'googleappmeasurement' 10.23.0

### ISSUE THREAD

**Issue by marinayxh** [firebase/firebase-ios-sdk#12554]
### Description

**Setting (Android Studio with Flutter)**:
<img width="899" alt="Screenshot 2024-03-14 at 2 59 22 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/54040070/76abb596-481e-4f5e-b97f-e079728b596e">
<img width="504" alt="Screenshot 2024-03-14 at 4 03 44 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/54040070/6dbfd130-06a0-4f2c-a4ad-a4d2e6e49891">

Error occurred when I try to run main.dart on iOS simulator. **And this is the error message**:
<img width="1551" alt="Screenshot 2024-03-14 at 3 01 42 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/54040070/a8600e9d-15e6-435f-88f9-db3b28175bf8">

I just installed the newest version of Xcode (version 15.3) three days ago, because when I tried to push the app to Apple Store, I received this warning saying my app was build on old version of iOS sdk, and it was gonna expire soon. I also updated Mac system to the newest version. After the updating, the first error I got is `xcodebuild: error: Could not resolve package dependencies:
  could not find a branch named ‘master’ in https://github.com/firebase/firebase-ios-sdk (did you mean ‘main’?)`. I fixed this by changing branch name in ./ios/Runner.xcodeproj/project.pbxproj. And then I got this googleappmeasurement problem.

I tried to clean cache, reinstall cocoapods but didn't work. I tried to update firebase to the latest version of 10.23.0, but then this googleappmeasurement issue occurred again. Any help is appreciated! Thank you all!

### Firebase SDK Version

10.12.0

### Xcode Version

15.3

### Installation Method

CocoaPods

### Firebase Product(s)

Authentication, Database, Firestore, Performance

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ncooke3** [comment]
That error indicates that Swift Package Manager is being used somewhere and is pointed to Firebase's `main` branch, as opposed to a released tag. The issue template mentioned that your project is using CocoaPods, is that correct? Either CocoaPods or Swift Package Manager should be used, but not both. And, if Swift Package Manager is used, it should not be getting Firebase from the `main` branch.

**Comment by marinayxh** [comment]
> That error indicates that Swift Package Manager is being used somewhere and is pointed to Firebase's `main` branch, as opposed to a released tag. The issue template mentioned that your project is using CocoaPods, is that correct? Either CocoaPods or Swift Package Manager should be used, but not both. And, if Swift Package Manager is used, it should not be getting Firebase from the `main` branch.

Thank you for your reply!!! 

Sorry, I'm confused. I'm new to this. So this "googleappmeasurement 10.23.0 version not found" issue is because I installed both swift package manager and cocoapods and I'm using them together (which is not right)? I found this from /ios/Runner.xcodeproj/project.pbxproj (not sure if this is helpful):
<img width="1099" alt="Screenshot 2024-03-14 at 4 52 34 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/54040070/14afe2db-655b-4e0b-b1a8-81ceecbc47ab">


It looks like it's getting firebase from swiftpackage, Hmm not sure what I should do next.

**Comment by ncooke3** [comment]
Sorry, I'm confused. I'm new to this. So this "googleappmeasurement 10.23.0 version not found" issue is because I installed both swift package manager and cocoapods and I'm using them together (which is not right)? I found this from /ios/Runner.xcodeproj/project.pbxproj (not sure if this is helpful):

Correct, using both package manager can lead to issues. Is it possible for you to open the project in Xcode? Manually editing the xcodeproj file can leading to issues. Assuming you use CocoaPods and want to stay with it, I think the solution is to remove the SwiftPM package dependency in Xcode.
